### PR TITLE
Fixed issue #2526

### PR DESCRIPTION
**Pull request by tomolimo**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #2526 

added unit tests

### CODE REVIEW

File: inc/savedsearch_user.class.php [hunk 1]
```diff
@@ -81,4 +81,28 @@ static function getSpecificValueToSelect($field, $name = '', $values = '', array
...
+    * Summary of getDefault
+    * @param mixed $users_id id of the user
+    * @param mixed $itemtype type of item
+    * @return array|boolean same output than SavedSearch::getParameters()
+    * @since version 9.2
+    */
+   static function getDefault($users_id, $itemtype) {
+      global $DB;
+
+      $query = "SELECT `savedsearches_id`
```
Comment (trasher) [inline inc/savedsearch_user.class.php]: Use dbiterator ;)

File: inc/savedsearch_user.class.php [hunk 2]
```diff
@@ -81,4 +81,28 @@ static function getSpecificValueToSelect($field, $name = '', $values = '', array
...
+                  FROM `glpi_savedsearches_users`
+                  WHERE `users_id`='$users_id'
+                        AND `itemtype` = '$itemtype'";
+      if (($result = $DB->query($query)) && ($DB->numrows($result) > 0)) {
+         $IDtoload = $DB->result($result, 0, 0);
+         // Load default bookmark for this $itemtype
+         $bookmark = new SavedSearch();
+         // Only get datas for bookmarks
```
Comment (trasher) [inline inc/savedsearch_user.class.php]: data is invariable

File: inc/search.class.php [hunk 3]
```diff
@@ -5123,9 +5123,14 @@ static function manageParams($itemtype, $params = [], $usesession = true,
           && class_exists($itemtype)
           && method_exists($itemtype, 'getDefaultSearchRequest')) {
 
-         $default_values = array_merge($default_values,
-                                       call_user_func([$itemtype,
-                                                            'getDefaultSearchRequest']));
+         $user_default_values = SavedSearch_User::getDefault(Session::getLoginUserID(), $itemtype);
+
+         if (!$user_default_values) {
+            // user has no default bookmark (nor public, nor private) for $temtype
```
Comment (trasher) [inline inc/search.class.php]: $temtype :)

File: inc/savedsearch_user.class.php (hunk 1 above)
Comment (tomolimo) [reply inc/savedsearch_user.class.php]: even if I just need one record?

File: inc/savedsearch_user.class.php (hunk 2 above)
Comment (tomolimo) [reply inc/savedsearch_user.class.php]: :smile:

File: inc/search.class.php (hunk 3 above)
Comment (tomolimo) [reply inc/search.class.php]: :smile:

**Comment by trasher** [comment]
Perfect, thanks for the changes : )

I've not yet tested the PR; I'll try to do that quickly so it can be approved when tests will be add.

**Comment by trasher** [comment]
I've just tested... And this does not work :(

On a clean base with test data, I've added a global bookmark on a computer search "pc2" that gives 2 results. When I click on the reset button; I see again all 6 computers.

**Comment by tomolimo** [comment]
Did you set this bookmark as default?

**Comment by trasher** [comment]
Yes, it was set ad default :/

**Comment by tomolimo** [comment]
Did you also make a test on Tickets?

**Comment by tomolimo** [comment]
I can reproduce this behavior for Computers, but for Tickets it's ok :(
I'm going to fix this this morning.

**Comment by trasher** [comment]
Weird... :) No, I've just tested on computers, anything else :)

**Pull request merged**